File: 30 Days of Code/Day 20 Sorting/Bobble.c
#include <stdio.h>

int BobbleSort(int *arr, int size);

int main()
{
	int n = 0;
	scanf("%d", &n);

	int array[n];

	for (int i = 0; i < n; i++) {
		scanf("%d", &array[i]);
	}

	int swap = BobbleSort(array, n);

	printf("Array is sorted in %d swaps.\n", swap);
	printf("First Element: %d\n", array[0]);
	printf("Last Element: %d\n", array[n-1]);
	return 0;
}

int BobbleSort(int *arr, int size) {
	int swap = 0;
	while (size-->1) {
		for (int i = 0; i < size; i++) {
			if (arr[i] > arr[i+1]) {
				int temp = arr[i];
				arr[i] = arr[i+1];
				arr[i+1] = temp;
				swap++;
			}
		}
	}
	return swap;
}

File: 30 Days of Code/Day 20 Sorting/Bobble.py
class BobbleSort():
    def __init__(self, arr):
        self.arr = arr

    def Sort(self):
        unsort = len(self.arr)
        swap_times = 0

        for item in reversed(range(unsort)):
            for i in range(item):
                if self.arr[i] > self.arr[i+1]:
                    temp = self.arr[i]
                    self.arr[i] = self.arr[i+1]
                    self.arr[i+1] = temp
                    swap_times += 1

        return swap_times

n = int(input())
array = list(map(int, input().split()))

bobble = BobbleSort(array)
swap = bobble.Sort()

print("Array is sorted in " + str(swap) + " swaps.")
print('First Element: ' + str(array[0]))
print('Last Element: ' + str(array[-1]))